src/arrayMethodSort.js

```diff
@@ -1,12 +1,83 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const undefinedArr = [];
+
+    for (let i = 0; i < this.length; ) {
+      if (this[i] === null) {
+        undefinedArr.unshift(...this.splice(i, 1));
```

Comment:
Using `unshift` here is inefficient as it shifts all elements of the array each time. Consider using `push` and reversing the array later if order matters.